### DISCUSSION TITLE
When to report the data

### DISCUSSION THREAD

**Discussion by ttyyzzqq** [apache/skywalking#11808] [Q&A]
Does SkyWalking have to wait for a request to end before reporting data? If a request takes thirty minutes to complete, does it have to be thirty minutes later to analyze?

**Comment by wu-sheng** [comment]
For Java and Golang, yes. It depends on the agent implementation.
Basically, a request should not last so long.
30min is a batch process task, which should run async, rather than through a request.

**Reply by ttyyzzqq** [reply]
Okay. Thank you.
